### ISSUE TITLE
canBeCreatedFromFormat wrongly returns false

### ISSUE THREAD

**Issue by Propaganistas** [briannesbitt/Carbon#2326]
Hello,

I encountered an issue with the following code:
```php
$date = '2021-05-03T00:00:00+02:00';
Carbon::canBeCreatedFromFormat($date, 'Y-m-d\TH:i:sp');
```

Carbon version: **2.46.0**

PHP version: **8.0.3**

I expected to get:

```
true
```

But I actually get:

```
false
```

Moreover, if I try to "reformat" the `$date` variable to the format used in the check, it returns exactly the same string. So it seems the format of the `$date` variable is fine...
```php
Carbon::parse($date)->format('Y-m-d\TH:i:sp') === $date;

// true
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

`p` is missing in the list, meanwhile you can use `P` instead.

**Comment by kylekatarnls** [comment]
Fixed for 2.48.0 release.

**Comment by Propaganistas** [comment]
Thanks for your quick response!
